src/main/java/org/apache/sysds/runtime/controlprogram/federated/FederationMap.java

```diff
@@ -384,6 +384,21 @@ private static void setThreadID(long tid, FederatedRequest[]... frsets) {
 				Arrays.stream(frset).forEach(fr -> fr.setTID(tid));
 	}
 
+	public void reverseFedMap() {
```

Comment:
This assumes that all federated maps are always sorted in order. I think that is not a constraint that we have.
Furthermore this would break if we have a federated mapping that is the custom variant, with ranges all over the place.

I think maybe that you actually don't want to modify the mapping at all like this, or make sure that you sort the ranges before attempting this.